include/amici_exception.h

```diff
@@ -7,9 +7,20 @@
 #include <cstdio>
 #include <cstring>
 #include <sstream>
-#include <execinfo.h>
-#include <dlfcn.h>    // for dladdr
-#include <cxxabi.h>   // for __cxa_demangle
+#define PLATFORM_WINDOWS 1
+#define PLATFORM_UNIX 2
+#if defined(_WIN32)
+    #define PLATFORM PLATFORM_WINDOWS // Windows
```

Comment:
Do we need the extra `PLATFORM`? Why not 
```
#if defined(_WIN32)
    #define PLATFORM_WINDOWS
#endif
...
```
Simplifies downstream usage to ` #ifdef PLATFORM_WINDOWS`